spec/requests/articles_spec.rb

```diff
@@ -1,11 +1,78 @@
 require 'rails_helper'
 
 RSpec.describe "Articles", type: :request do
+
+  before(:all) do
+    FactoryBot.create(:article)
+  end
+
   describe "GET article index" do
-    it "returns http success" do
-      get "/articles#index"
+    it "returns success with the articles index template" do
+      get articles_path
+
+      expect(response).to have_rendered(:index)
       expect(response).to have_http_status(:success)
     end
   end
 
+  describe "GET /articles/:id" do
+    it "returns success with the articles index template" do
+      get article_path(1)
+
+      expect(response).to have_rendered(:show)
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "GET /articles/new" do
+    it "returns success with the articles/new template" do
+      get new_article_path
+
+      expect(response).to have_rendered(:new)
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe "POST /articles" do
+    context "with valid parameters" do
+      it "successfully creates an article" do
+        post articles_path, :params => {:article => {:title => Faker::Hipster.sentence(word_count: 3), :body => Faker::Hipster.paragraph}}
+
+        expect(response).to have_http_status(:found)
+      end
+    end
+
+    context "with invalid parameters" do
+      it "responds with unprocessable entity" do
+        post articles_path, :params => {:article => {:title => "", :body => "Foo"}}
+
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+    end
+  end
+
+  describe "PUT /articles/:id" do
```

Comment:
What happens if we try and update an article that doesn't exist? These tests are about the parameters being passed on the body. Do we get a 404 back if we try and update an article that doesn't exist?

Reply:
Good catch, definitely a missing case here. If we pass an id that doesn't exist, it just fails on the find and throws an exception. In this case would I be writing a test to test that the exception `ActiveRecord::RecordNotFound` is thrown?